lib/wiki.ml

```diff
@@ -108,17 +110,108 @@ let str_start_with str i x =
   in
   loop i 0
 
+let escape (s : string) = (Util.escape_html s : Adef.escaped_string :> string)
+let encode (s : string) = (Mutil.encode s : Adef.encoded_string :> string)
+
 type wiki_info = {
   wi_mode : string;
   wi_file_path : string -> string;
-  wi_person_exists : string * string * int -> bool * string * string;
+  wi_person_exists : string * string * int -> bool;
+  wi_mark_if_not_public : string * string * int -> bool;
   wi_always_show_link : bool;
 }
 
-let escape (s : string) = (Util.escape_html s : Adef.escaped_string :> string)
-let encode (s : string) = (Mutil.encode s : Adef.encoded_string :> string)
+let find_first_char_from_list str i chars =
+  let len = String.length str in
+  let rec find_index j =
+    if j >= len then None
+    else if List.mem str.[j] chars then Some j
+    else find_index (j + 1)
+  in
+  find_index i
+
+let bold_italic_syntax s =
+  let chars = [ '{'; '%'; '\'' ] in
+  let slen = String.length s in
+  let rec loop quot_lev i len =
+    let len, quot_lev =
+      if i = slen || List.exists (str_start_with s i) [ "</li>"; "</p>" ] then
+        let len =
+          match quot_lev with
+          | 1 -> Buff1.mstore len "</i>"
+          | 2 -> Buff1.mstore len "</b>"
+          | 3 -> Buff1.mstore len "</b></i>"
+          | _ -> len
```

Comment:
Using integers to store the state of the automaton is harder to read. An enum type is better for readability.

Reply:
done